### PR TITLE
[webgpu] Fix sum operator

### PR DESCRIPTION
**Pull request by haoyunfeix**
Set axes correct value when input is transposed

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4787)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@qjia7 @axinging @xhcao @gyagp PTAL

**Comment by xhcao** [comment]
Is there the same issue of `max` and `min` operators?

**Comment by qjia7** [comment]
> Is there the same issue of `max` and `min` operators?

I guess it will have the similar issue with `max` and `min`.  But I prefer you directly refactor these parts in #4782. With that, the bugs in this patch can be fixed together.

**Review by xhcao — COMMENTED** [review]
> > Is there the same issue of `max` and `min` operators?
> 
> I guess it will have the similar issue with `max` and `min`. But I prefer you directly refactor these parts in #4782. With that, the bugs in this patch can be fixed together.

Ok, I will try to refactor this code and fix these bugs.

File: tfjs-backend-webgpu/src/kernels/Sum.ts [hunk 1]
```diff
@@ -34,7 +34,13 @@ export function sum(
...
 
   const origAxes = util.parseAxisParam(axis, xShape);
-  const axes = origAxes;
+  let axes = origAxes;
+  const permutedAxes = backend_util.getAxesPermutation(axes, xRank);
+  const sumInputIsTransposed = permutedAxes != null;
+
+  if (sumInputIsTransposed) {
+    axes = backend_util.getInnerMostAxes(axes.length, xRank);
+  }
```
[the remark this answers is not in the document]
Comment (haoyunfeix) [reply tfjs-backend-webgpu/src/kernels/Sum.ts:47]: Done. Please take another look, thanks!

[the remark this answers is not in the document]
Comment (xhcao) [reply tfjs-backend-webgpu/src/kernels/Sum.ts:47]: I had refactered all reduce related operators in #4782, so we may not need this CL.

[the remark this answers is not in the document]
Comment (haoyunfeix) [reply tfjs-backend-webgpu/src/kernels/Sum.ts:47]: That's Great!

**Comment by haoyunfeix** [comment]
This is covered by https://github.com/tensorflow/tfjs/pull/4782, close this one.

**Pull request closed without merging**